diff --git a/src/devices/acurite.c b/src/devices/acurite.c
index 08c555aa1..5c1b6072f 100644
--- a/src/devices/acurite.c
+++ b/src/devices/acurite.c
@@ -996,6 +996,13 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
                 valid++;
             }
             else if (message_type == ACURITE_MSGTYPE_RAINFALL) {
+                // The earlier length check passes as ACURITE_TXR_BITLEN is 56
+                // bits / 7 bytes. This message type is always 8 bytes.
+                if (browlen != 8) {
+                    decoder_log(decoder, 2, __func__, "skipping wrong len");
+                    continue;
+                } // DECODE_ABORT_LENGTH
+
                 // Rain Fall Gauge 899
                 // The high 2 bits of byte zero are the channel (bits 7,6), 00 = A, 01 = B, 10 = C
                 int channel = bb[0] >> 6;
