From 29c04d8b412eb09f4cd1db4b363050d2d72e9065 Mon Sep 17 00:00:00 2001
From: Peter Bakhirev <pbkhrv@pm.me>
Date: Wed, 17 Mar 2021 20:41:26 -0400
Subject: [PATCH] Remove invalid device_classes: "safety"

---
 examples/rtl_433_mqtt_hass.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/examples/rtl_433_mqtt_hass.py b/examples/rtl_433_mqtt_hass.py
index f710985f..d672c2f0 100755
--- a/examples/rtl_433_mqtt_hass.py
+++ b/examples/rtl_433_mqtt_hass.py
@@ -334,7 +334,6 @@ mappings = {
         "device_type": "binary_sensor",
         "object_suffix": "tamper",
         "config": {
-            "device_class": "safety",
             "force_update": "true",
             "payload_on": "1",
             "payload_off": "0"
@@ -345,7 +344,6 @@ mappings = {
         "device_type": "binary_sensor",
         "object_suffix": "alarm",
         "config": {
-            "device_class": "safety",
             "force_update": "true",
             "payload_on": "1",
             "payload_off": "0"
-- 
2.25.1

